src/coreComponents/linearAlgebra/interfaces/BlasLapackLA.cpp

```diff
@@ -789,4 +783,76 @@ void BlasLapackLA::matrixSVD( arraySlice2d< real64 const, MatrixLayout::ROW_MAJO
   }
 }
 
+void BlasLapackLA::matrixEigenvalues( MatColMajor< real64 const > const & A,
+                                      Vec< std::complex< real64 > > const & lambda )
+{
+  GEOSX_ASSERT_MSG( A.size( 0 ) == A.size( 1 ),
+                    "The matrix A must be square" );
+
+  GEOSX_ASSERT_MSG( A.size( 0 ) == lambda.size(),
+                    "The matrix A and lambda have incompatible sizes" );
+
+  // make a copy of A, since dgeev destroys contents
+  array2d< real64, MatrixLayout::COL_MAJOR_PERM > ACOPY( A.size( 0 ), A.size( 1 ) );
+  BlasLapackLA::matrixCopy( A, ACOPY );
+
+  // define the arguments of dgesvd
+  int const N    = LvArray::integerConversion< int >( A.size( 0 ) );
+  int const LDA  = N;
+  int const LDVL = 1;
+  int const LDVR = 1;
+  int LWORK = 0;
+  int INFO  = 0;
+  double WKOPT = 0.0;
+  double VL = 0.0;
+  double VR = 0.0;
+
+  array1d< real64 > WR( N );
+  array1d< real64 > WI( N );
+
+  // 1) query and allocate the optimal workspace
+  LWORK = -1;
+  GEOSX_dgeev( "N", "N",
+               &N, ACOPY.data(), &LDA,
+               WR.data(), WI.data(),
+               &VL, &LDVL,
+               &VR, &LDVR,
+               &WKOPT, &LWORK, &INFO );
+
+  LWORK = static_cast< int >( WKOPT );
+  array1d< real64 > WORK( LWORK );
+
+  // 2) compute eigenvalues
+  GEOSX_dgeev( "N", "N",
+               &N, ACOPY.data(), &LDA,
+               WR.data(), WI.data(),
+               &VL, &LDVL,
+               &VR, &LDVR,
+               WORK.data(), &LWORK, &INFO );
+
+  for( int i = 0; i < N; ++i )
+  {
+    lambda[i] = std::complex< real64 >( WR[i], WI[i] );
+  }
+
+  GEOSX_ASSERT_MSG( INFO == 0, "The algorithm computing eigenvalues failed to converge." );
```

Comment:
This might be how `BlasLapackLA` does it but these should probably be converted to `GEOSX_ERROR_IF` since the overhead is tiny and at least convergence errors are most likely to come up in large (release build) runs.

Reply:
You're right. Thanks!